### PR TITLE
refactored gamescore so driver is less hardcoded

### PR DESCRIPTION
**Pull request by seanheinrichs**

### CODE REVIEW

File: src/GameScore/GameScore.cpp [hunk 1]
```diff
@@ -7,258 +6,231 @@
 
 GameScore::GameScore() {}
 
-GameScore::~GameScore() {
-
-}
+GameScore::~GameScore() {}
```
Comment (mdesson) [inline src/GameScore/GameScore.cpp]: Could we define this or remove it entirely? If there's nothing to clean up I think we can just the one provided by the compiler.

File: src/GameScore/GameScore.cpp [hunk 2]
```diff
@@ -7,258 +6,231 @@
 
 GameScore::GameScore() {}
```
Comment (mdesson) [inline src/GameScore/GameScore.cpp]: Could we define this or remove it entirely? If there's nothing to specifically inititalize I think we can just the one provided by the compiler.

File: src/GameScore/GameScore.h [hunk 3]
```diff
@@ -8,56 +7,26 @@
...
+    int countTreePoints(int tree);
+    int countAnvilPoints(int anvil);
+    int countCarrotPoints(int carrot);
+    int countRockPoints(int rock);
+    int countCrystalPoints(int crystal);
 
-//    std::vector<Cards*> players->getGameHand();
 public:
+    // constructors
     GameScore();
```
Comment (mdesson) [inline src/GameScore/GameScore.h]: I think we can safely remove this.

File: src/GameScore/GameScore.h [hunk 4]
```diff
@@ -8,56 +7,26 @@
...
 
-    GameScore(const vector<Cards *> &gameHand, const vector<Player *> &players);
-
+    // destructor
     virtual ~GameScore();
```
Comment (mdesson) [inline src/GameScore/GameScore.h]: I think we can safely remove this.

File: src/GameScore/GameScoreDriver.cpp [hunk 5]
```diff
@@ -8,10 +8,8 @@
 
 void gameScoreExample() {
     MapLoader loader;
-    std::string fileLocation = "../assets/map1.map";
+    std::string fileLocation = "assets/map1.map";
```
Comment (mdesson) [inline src/GameScore/GameScoreDriver.cpp]:
```suggestion
    std::string fileLocation = "assets/map1.map"; // in CLion add ../ to start of string
```
Might be worth adding a comment.

**Review by mdesson — CHANGES_REQUESTED** [review]
Over all it looks good! Main recommendations would be to just remove the useless, empty constructors and destructors.

File: src/GameScore/GameScore.cpp (hunk 1 above)
Comment (seanheinrichs) [reply src/GameScore/GameScore.cpp]: Yup, good call. Removed!

File: src/GameScore/GameScore.h (hunk 3 above)
Comment (seanheinrichs) [reply src/GameScore/GameScore.h]: Yup, removed!

File: src/GameScore/GameScore.h (hunk 4 above)
Comment (seanheinrichs) [reply src/GameScore/GameScore.h]: Removed!

File: src/GameScore/GameScoreDriver.cpp (hunk 5 above)
Comment (seanheinrichs) [reply src/GameScore/GameScoreDriver.cpp]: Definitely, added a separate string so we can just quickly uncomment if using a different IDE. Although I'm finding myself doing more C++ in VSCODE now that I know how to compile it haha.

**Pull request merged**